src/cache/cache.go

```diff
@@ -67,16 +68,36 @@ func (c *Cache) Init(id int, cacheSize int, cacheType config.CacheType, data *da
 	}
 }
 
+func (c* Cache) Killed() bool {
+    return !c.alive
```

Comment:
Hey @meiaalsup if we're locking to `Kill()` and also to `Revive()`, we should also lock to check for `Killed()`

Reply:
I see that the only place you used `Killed()` in `cache.go` is in `Fetch()` on line `95`, so it *is* fine to not lock here. 

But since `Killed()` looks like an externally facing method, it should lock at least when externally accessed

Reply:
fixed :)